components/sidebars/SidebarEditTags.js

```diff
@@ -0,0 +1,160 @@
+import * as React from "react";
+import * as Strings from "~/common/strings";
+import * as Constants from "~/common/constants";
+import * as System from "~/components/system";
+import * as Validations from "~/common/validations";
+import * as Actions from "~/common/actions";
+import * as Events from "~/common/custom-events";
+
+import { css } from "@emotion/react";
+import { Tag } from "~/components/system/components/Tag";
+
+const STYLES_GROUPING = css`
+  width: 100%;
+  border: 1px solid rgba(196, 196, 196, 0.5);
+  background-color: ${Constants.system.white};
+  border-radius: 6px;
+  padding: 16px;
+  margin-bottom: 40px;
+`;
+
+export default class SidebarEditTags extends React.Component {
+  state = {
+    tags:
+      !Array.isArray(this.props.sidebarData.commonTags) ||
+      this.props.sidebarData.commonTags?.length === 0
+        ? []
+        : this.props.sidebarData.commonTags,
+    suggestions:
+      !Array.isArray(this.props.sidebarData.suggestions) ||
+      this.props.sidebarData.suggestions?.length === 0
+        ? []
+        : this.props.sidebarData.suggestions,
+  };
+
+  _handleChange = (e) => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  updateSuggestions = () => {
+    let newSuggestions = new Set([...this.props.sidebarData.suggestions, ...this.state.tags]);
+    this.setState({ suggestions: Array.from(newSuggestions) });
+  };
+
+  _handleSave = async (details, index) => {
+    let { objects } = this.props.sidebarData;
+
+    objects[index] = { ...objects[index], ...details };
+    const response = await Actions.updateData({
+      id: this.props.viewer.id,
+      data: objects[index],
+    });
+
+    this.props.onUpdateViewer({ tags: this.state.suggestions });
+
+    Events.hasError(response);
+  };
+
+  _handleSubmit = async () => {
+    this.props.onCancel();
+
+    let { checked, objects, commonTags } = this.props.sidebarData;
+    const checkedIndexes = Object.keys(checked);
+    let data;
+
+    await Promise.all(
```

Comment:
I suspect the reason some are updating and not others is becuase you are doing a promise.all (aka make all these calls to the server at once). Because browsers are single threaded (only do one thing at a time), trying to do this means only some go through and others don't.

What you should be doing (which is much more efficient) is making a new endpoint for the api which can handle updating multiple files at once and then just make a single call to that. You can make a new one and base it off the existing api/data/update.js

Reply:
Thank you so much for this!! I decided to modify `api/data/update.js` to handle an array of updates.